assets/js/modules/idea-hub/datastore/idea-state.js

```diff
@@ -0,0 +1,133 @@
+/**
+ * `modules/idea-hub` data store: idea-state.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+import isPlainObject from 'lodash/isPlainObject';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from './constants';
+import { createFetchStore } from '../../../googlesitekit/data/create-fetch-store';
+
+const fetchPostUpdateIdeaStateStore = createFetchStore( {
+	baseName: 'updateIdeaState',
+	controlCallback: () => {
+		return API.set( 'modules', 'idea-hub', 'update-idea-state' );
+	},
+} );
+
+const baseInitialState = {};
+
+const baseActions = {
+	/**
+	 * Updates a given Idea's state.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object}  ideaState           Idea Hub Idea state.
+	 * @param {string}  ideaState.name      Idea Hub Idea name.
+	 * @param {boolean} ideaState.saved     Whether the Idea is saved [optional].
+	 * @param {boolean} ideaState.dismissed Whether the Idea is dismissed [optional].
+	 * @return {Object} Object with `response` and `error`.
+	 */
+	*updateIdeaState( ideaState ) {
+		invariant( isPlainObject( ideaState ), 'ideaState must be an object.' );
+		invariant( typeof ideaState.name === 'string', 'ideaState.name must be a string.' );
+
+		const response = yield fetchPostUpdateIdeaStateStore.actions.fetchUpdateIdeaState( ideaState );
+
+		return response;
+	},
+	/**
+	 * Saves an Idea.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string} ideaName Idea Hub Idea name.
+	 * @return {Object} Object with `response` and `error`.
+	 */
+	*saveIdea( ideaName ) {
+		invariant( typeof ideaName === 'string', 'ideaName must be a string.' );
+
+		const registry = yield Data.commonActions.getRegistry();
+		const response = yield registry.dispatch( STORE_NAME ).updateIdeaState( {
+			name: ideaName,
+			saved: true,
+		} );
```

Comment:
No need to get the registry and dispatch an additional action here. You can yield the fetch action directly. Plus, you need to correctly process possible errors in response. See an example here: https://github.com/google/site-kit-wp/blob/70f289814bddbad345ba26c48ab25666c52db1e0/assets/js/modules/analytics/datastore/accounts.js#L212-L217

Reply:
@eugene-manuilov Maybe I misunderstood the ACs, I thought these actions were all supposed to call the `updateIdeaState` action internally. 
If they can just do `fetchPostUpdateIdeaStateStore.actions.fetchUpdateIdeaState( ideaState );` then why have `updateIdeaState`?
Have I misunderstood?

Reply:
Ah... yes, you are right, no need to call it directly then. We can continue calling `updateIdeaState` but we need to hoist errors up to the called action using `clearError`/`receiveError` helpers.